src/embed_tests/pyinitialize.cs

```diff
@@ -24,10 +24,12 @@ public static void StartAndStopTwice()
         public static void LoadDefaultArgs()
         {
             using (new PythonEngine())
-            using (var argv = new PyList(Runtime.Runtime.PySys_GetObject("argv")))
             {
-                Runtime.Runtime.XIncref(argv.Handle);
-                Assert.AreNotEqual(0, argv.Length());
+                var argvref = new BorrowedReference(Runtime.Runtime.PySys_GetObject("argv"));
```

Comment:
NIT: I believe `PySys_GetObject` is rarely used. It's return type can be changed to `BorrowedReference`.

Reply:
It's used around 15 times in the codebase and most places use the IntPtr. Best be done a separate PR

Reply:
You removed XIncref which means your reference count will be incorrect.
Why make these changes, there's no bugs here.

Reply:
Ah my bad. Thinking about it, why do we need to increment the refcount of a borrowed reference? I added a new constructor to make `PyList` from a `BorrowedReference`. The list does take ownership of it (and decrefs it when Disposed)